src/Helper/FieldManager.php

```diff
@@ -0,0 +1,243 @@
+ *
+ * Automatically handles required relations based on the field format: entity.field. Relations can be chained by
+ * entity.entity.field.
+ */
+class FieldManager
+{
+    const ROOT_ALIAS = 'r';
+
+    /**
+     * Added fields.
+     *
+     * @var array
+     */
+    protected $fields = [];
+
+    /**
+     * Metadata for fields, per entity.
+     *
+     * @var array
+     */
+    protected $entityFieldMetaData = [];
+
+    /**
+     * Relation information.
+     *
+     * @var array
+     */
+    protected $relations = [];
+
+    /**
+     * @var string
+     */
+    protected $rootEntity;
+
+    /**
+     * @var EntityManager
+     */
+    protected $entityManager;
+
+    /**
+     * Gets the EntityManager.
+     *
+     * @return EntityManager
+     *   The EntityManager.
+     */
+    public function getEntityManager(): EntityManager
+    {
+        return $this->entityManager;
+    }
+
+    /**
+     * Sets the EntityManager.
+     *
+     * @param EntityManager $entityManager
+     *   The EntityManager.
+     */
+    public function setEntityManager(EntityManager $entityManager): void
+    {
+        $this->entityManager = $entityManager;
+    }
+
+    /**
+     * Gets the RootEntity.
+     *
+     * @return string
+     *   The RootEntity.
+     */
+    public function getRootEntity(): string
+    {
+        return $this->rootEntity;
+    }
+
+    /**
+     * Sets the RootEntity.
+     *
+     * @param string $rootEntity
+     *   The RootEntity.
+     */
+    public function setRootEntity(string $rootEntity): void
+    {
+        $this->rootEntity = $rootEntity;
+    }
+
+    /**
+     * Add a field to the fieldmanager.
+     *
+     * @param $fieldName
+     *
+     * @return array
+     *   Field information with relation alias.
+     *
+     * @throws EntityNotFoundException
+     */
+    public function addField($fieldName) {
+        if (!empty($this->fields[$fieldName])) {
+            return $this->fields[$fieldName];
+        }
+
+        $this->fields[$fieldName] = $this->parseField($fieldName);
+
+        $this->setRelations($fieldName);
+        $relation = $this->relations[$this->fields[$fieldName]['entity']];
+        $this->fields[$fieldName]['relation_alias'] = $relation['alias'];
+
+        $this->fields[$fieldName]['metadata'] = $this->getFieldMetaData($this->fields[$fieldName]['entity'], $this->fields[$fieldName]['field']);
+
+        return $this->fields[$fieldName];
+    }
+
+    /**
+     * @param string $fieldMetadata
+     *
+     * @return string
+     */
+    public function getQueryFieldName(string $fieldName): string
+    {
+
+        return sprintf(
+            '%s.%s',
+            $this->fields[$fieldName]['relation_alias'] ?? FieldManager::ROOT_ALIAS,
+            $this->fields[$fieldName]['field']
+        );
+    }
+
+    /**
+     * Get the field data for the requested field.
+     *
+     * @param $fieldName
+     *
+     * @return array|null
+     */
+    public function getField($fieldName) {
```

Comment:
please follow rules of PSR-2 and also define the return type

```suggestion
    public function getField($fieldName) : ?array
    {
```